File: solutions/_557_reverse_words_in_a_string_iii/solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution
{
public:
    string reverseWords(string s)
    {
        vector<string> words;
        string word = "";
        for (int i = 0; i < s.size(); i++)
        {
            if (s[i] == ' ')
            {
                words.push_back(word);
                word = "";
            }
            else
            {
                word += s[i];
            }
        }
        words.push_back(word);
        string result = "";
        for (int i = 0; i < words.size(); i++)
        {
            reverse(words[i].begin(), words[i].end());
            result += words[i];
            if (i < words.size() - 1)
            {
                result += " ";
            }
        }
        return result;
    }
};

/*
https://leetcode.com/problems/reverse-words-in-a-string-iii/
*/

File: solutions/_557_reverse_words_in_a_string_iii/solution.java
package _557_reverse_words_in_a_string_iii;

class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        StringBuilder result = new StringBuilder();
        for (String word : words) {
            result.append(new StringBuilder(word).reverse().toString() + " ");
        }
        return result.toString().trim();
    }
}

/**
 * https://leetcode.com/problems/reverse-words-in-a-string-iii/
 */